lib/bolt/task/puppet_server.rb

```diff
@@ -11,19 +11,17 @@ def initialize(task_data, file_cache)
         super(task_data)
       end
 
-      # TODO: currently this both converts filename -> name and fetches files to generate the path
-      # In the future we should not generate paths until after the implementation is selected to avoid
-      # unecessary fetching of files
+      # puppetserver file entries have 'filename' rather then 'name'
```

Comment:
That's inconvenient.

Reply:
Yeah I discovered that a few months ago when working on cross platform tasks in puppetserver. It's a public API so we should probably change puppet.